pkg/ccl/streamingccl/replication_manager.go

```diff
@@ -61,3 +61,20 @@ func (r replicationStreamManagerImpl) UpdateReplicationStreamProgress(
 	}
 	return UpdateReplicationStreamProgressHook(evalCtx, streamID, frontier, txn)
 }
+
+// StreamPartitionHook hooks a StreamPartition implementation inside streamingccl package.
+var StreamPartitionHook func(
```

Comment:
I don't really understand why we have all these hooks and mutable function pointers in the ccl code.

Reply:
yeah, I think we need to clean this up.  There should be just 1 hook to set the manager.  Though, maybe the issue was w/ circular dependencies on jobs package? @gh-casper do you remember?